exampleCourse/questions/element/dragDropBlocks/question.html

```diff
@@ -0,0 +1,108 @@
+  Remember to indent your code! </p>
+<p>
+  This example uses the attribute <code>pl-hide-in-panel answer="true"</code> to hide the correct answer. It
+  also uses <code>permutation-mode="ranking"</code> to control how the correct solution is ordered
+  and <code>indent</code> attribute in <code>pl-answer</code> to define expected code indentation.
+</p>
+</pl-question-panel>
+
+<pl-hide-in-panel answer="true">
+<pl-order-blocks answers-name="python-ordered" shuffle-options="true"  permutation-mode="ranking">
+  <pl-answer correct="true" ranking="1" indent="1">def my_sum(first, second):</pl-answer>
+  <pl-answer correct="true" ranking="2" indent="2">sum = 0</pl-answer>
+  <pl-answer correct="true" ranking="3" indent="2">sum += first</pl-answer>
+  <pl-answer correct="true" ranking="3" indent="2">sum += second</pl-answer>
+  <pl-answer correct="true" ranking="4" indent="2">return sum</pl-answer>
+  <pl-answer correct="false">sum + first</pl-answer>
+  <pl-answer correct="false">sum + second</pl-answer>
+  <pl-answer correct="false">This is an example very long tile to test CSS overflow</pl-answer>
+
+</pl-order-blocks>
+</pl-hide-in-panel>
+
+
+<hr>
+
+
+<pl-question-panel>
+<p>
+  <strong>Example 4:</strong> Let $a, b$ be any even numbers. Prove that $a + b$ is always even.
+</p>
+
+<p>This example uses the attribute <code>permutation-mode="ranking"</code> to control how the correct solution is ordered.
+When the order of a correct block is not relevant,
+ the attribute <code>ranking</code> is not provided. In this example, two of the blocks can be placed at any
+position when constructing the solution. The attribute <code>dropzone-layout="verticalLayout"</code> is used
+to display the input area above the output area.</p>
+</pl-question-panel>
+
+ <pl-order-blocks answers-name="latex_answers" shuffle-options="true" permutation-mode="ranking" dropzone-layout="verticalLayout">
+  <pl-answer correct="true" ranking="1" >Proof by Contrapositive:</pl-answer>
+  <pl-answer correct="true" ranking="2" >Take some $m \in \mathbb{Z}$ </pl-answer>
+  <pl-answer correct="true" ranking="2" >Take some $n \in \mathbb{Z}$ </pl-answer>
+  <pl-answer correct="true" >Assume $m$ is even</pl-answer>
+  <pl-answer correct="true" >Assume $n$ is even</pl-answer>
+  <pl-answer correct="true" ranking="3" >$\exists l \in \mathbb{Z}$ such that $m = 2l$</pl-answer>
+  <pl-answer correct="true" ranking="3" >$\exists k \in \mathbb{Z}$ such that $n = 2k$</pl-answer>
+  <pl-answer correct="true" ranking="4">$am^2 + bm + cn^2 + dn + e$</pl-answer>
+  <pl-answer correct="true" ranking="5">$= a(2l)^2 + b(2l) + c(2k)^2 + d(2k) + e$ </pl-answer>
+  <pl-answer correct="true" ranking="6">$= 2(2al^2 + bl + 2ck^2 + dk) + e$ </pl-answer>
+  <pl-answer correct="true" ranking="7">Since $e$ is odd, $2(2al^2 + bl + 2ck^2 + dk) + e$ is odd </pl-answer>
+  <pl-answer correct="true" ranking="8">$QED$</pl-answer>
+</pl-order-blocks>
```

Comment:
Yeah, this is one of those CSS headaches
The `Proof by contrapositive` block is correctly displayed, because it's plain text and contains no latex. For any blocks with Latex, because the Latex are rendered using `svg`, the behaviour is extremely odd

Reply:
I'm not really understanding what's causing this problem. Mathjax renders the latex into whatever divs/spans are already around the math text, so surrounding blocks should be fine. For example, bootstrap containers work fine with mathjax inside them.

Maybe this is a reason to switch to using bootstrap styling for the blocks?

Reply:
Definitely, let's hope bootstrap will automagically fix this for us

Reply:
cleaning up redundant code also fixed this incorrect styling... somewhat
![image](https://user-images.githubusercontent.com/11191061/103158779-9b2c0b00-4776-11eb-8819-d2b7d356e0cd.png)
I'm not sure if you really like this styling of `inline-block`. My intend was to minimise screen space usage, because displaying blocks vertically would take up a lot of space.

Reply:
I agree that vertical display of blocks takes up more space, but I think it would be best to default to vertical display (that is, every block in it's own div so that they display vertically, and none of them are inline with the text "Correct answer (in the specified order):". This matches the display that the student was working with in the solution dropzone, so I think this is the best default for displaying the correct answer and submitted answers. We can defer this to the second PR, however (I added it to #3476).

In the future we could add an option to use inline display for the blocks. This would be nice in the interactive dropzones as well. For example, in the examples like "choose all the prime numbers and put them in order", it would make more sense for all the blocks to be inline (in the source area, the solution area, correct answer, submitted answer). I added this to #3476.